### ISSUE TITLE
Cannot parse directory of .ics files or parse some types of events in ics files

### ISSUE THREAD

**Issue by geekpete** [b1rger/carl#46]
I have config file loading .ics file working along with agenda working too.

I couldn't get directory mode of a bunch of `.ics` files to work, but have a basic birthday dates ICS file working fine by itself , so suspect some event types break parsing.
Testing directory mode with only that working ics file shows directory mode works fine as long as it can parse the ics files.

I'll can try to dig into which event types those are, of the 4 ics files I exported, only the simple birthday one works that has only 2 basic birthday events in it.

Probably related to: https://github.com/b1rger/carl/issues/32

Overall, a cool little utility that I'll probably use in my host MOTDs or login banner for interactive logins....**IF** I can find a way to sync my calendars out to ics files reliably...

Testing an unparsable ics file:
```
 RUST_BACKTRACE=all carl -y -a
thread 'main' panicked at 'not yet implemented', /home/peter/.cargo/registry/src/github.com-1ecc6299db9ec823/carl-0.1.0/src/events/mod.rs:71:39
stack backtrace:
   0: rust_begin_unwind
             at /build/rustc-ujz1Py/rustc-1.66.1+dfsg0ubuntu1/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /build/rustc-ujz1Py/rustc-1.66.1+dfsg0ubuntu1/library/core/src/panicking.rs:65:14
   2: core::panicking::panic
             at /build/rustc-ujz1Py/rustc-1.66.1+dfsg0ubuntu1/library/core/src/panicking.rs:115:5
   3: carl::events::Event::in_range
   4: alloc::vec::in_place_collect::<impl alloc::vec::spec_from_iter::SpecFromIter<T,I> for alloc::vec::Vec<T>>::from_iter
   5: carl::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Files were exported with Gmail export, but I can dig deeper into which event types break it if you want, or hold off if you're already aware of the parsing issue.

**Comment by b1rger** [comment]
Glad you like it!

Thanks for the detailed report - the `thread 'main' panicked at 'not yet implemented'` makes me think that this is about https://github.com/b1rger/carl/blob/cc24697b736ff1648cbdc5b2886800e7def15b37/src/events/mod.rs#L71

Can you confirm that there are events in the ics files that have a frequency of occuring weekly?

I was hoping that #32 might make ical parsing more comfortable, but it turned out not to be the case. I haven't decided yet if the switch makes sense. In any case, fixing the missing support for weekly events should be relatively easy.

**Comment by geekpete** [comment]
Some examples weekly frequency events:
`RRULE:FREQ=WEEKLY;UNTIL=20220728T135959Z;BYDAY=FR`
and
`RRULE:FREQ=WEEKLY;UNTIL=20220724T135959Z;BYDAY=FR,MO,TH,TU,WE`
and
`RRULE:FREQ=WEEKLY;WKST=SU;UNTIL=20220726T135959Z;BYDAY=FR,MO,TH,TU,WE`

And seems this part of the spec can be a bit complex:
- https://datatracker.ietf.org/doc/html/rfc2445#section-4.8.5.4
- https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html

**Comment by b1rger** [comment]
I've implemented a fix for the crash. `carl` does not yet parse recurring events correctly- you're totally right that `RRULE` can be quite complex. I've created #48 to evaluate one solution

**Comment by b1rger** [comment]
Just FTR, I've released `carl` 0.2.0 which now uses `rrule` for parsing the recurring events
